addons/mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -24,13 +24,82 @@
             var self = this;
             this._super();
             this.choose_mailing_list('click').fail(function () {
-                self.editor.on_remove();
+                self.editor.on_remove($.Event( "click" ));
             });
         },
         clean_for_save: function () {
             this.$target.addClass("hidden");
         },
     });
+
+    website.snippet.options.newsletter_popup = website.snippet.Option.extend({
+        select_mailing_list: function (type, value) {
+            var self = this;
+            if (type !== "click") return;
+            return website.prompt({
+                id: "editor_new_mailing_list_subscribe_popup",
+                window_title: _t("Add a Newsletter Subscribe Popup"),
+                select: _t("Newsletter"),
+                init: function (field) {
+                    return website.session.model('mail.mass_mailing.list')
+                            .call('name_search', ['', []]);
+                },
+            }).then(function (mailing_list_id) {
```

Comment:
hello @tde-banana-odoo , when drag and drop the snippet, open Dialog to ask for the newsletter selection , after selection of newsletter it will just return the selected newsletter ID(Just ID). Thanks